apps/website/docs/react/api/merge-classes.md

```diff
@@ -0,0 +1,89 @@
+---
+sidebar_position: 2
+---
+
+import OutputTitle from '@site/src/components/OutputTitle'
+
+# mergeClasses
+
+There are cases where you need to merge classes from multiple `useClasses` calls. To properly merge the classes, you need to use `mergeClasses()` function, which performs merge and deduplication of atomic classes generated by `makeStyles()`.
+
+```jsx
+import { mergeClasses, makeStyles } from '@griffel/react';
+
+const useClasses = makeStyles({
+  blueBold: {
+    color: 'blue',
+    fontWeight: 'bold',
+  },
+  red: {
+    color: 'red',
+  },
+});
+
+function Component(props) {
+  const { isBold } = props;
+  const classes = useClasses();
+
+  // ℹ️ Order of arguments determines the results
+
+  const firstClassName = mergeClasses(isBold && classes.blueBold, classes.red);
+  // 👆 { color: 'red', fontWeight: 'bold' }
+  const secondClassName = mergeClasses(classes.red, isBold && classes.blueBold);
+  // 👆 { color: 'blue', fontWeight: 'bold' }
+
+  return (
+    <>
+      <div className={firstClassName} />
+      <div className={secondClassName} />
+    </>
+  );
+}
+```
+
+:::note
+
+Compared to native CSS order of arguments in `mergeClasses()` determines the results. This allows to control styles with JavaScript.
+
+:::
+
+## Incorrect usages
+
+This section shows and describes anti-patterns which should never be used.
+
+### Never concatenate class strings
+
+:::caution
```

Comment:
Let' s also give a code sample for merging without `mergeClasses`

Reply:
Not sure that I get this, we have a sample below that shows wrong behavior:

![image](https://user-images.githubusercontent.com/14183168/156205096-d4f96a4f-9f81-4c97-b4a4-3725d4673f48.png)

Reply:
I guess, my point is that currently the section is just saying "dont't merge classes without `mergeClasses`".

I'm missing the why

Reply:
Added a note why.